src/components/material.js

```diff
@@ -176,20 +177,38 @@ module.exports.Component = registerComponent('material', {
    */
   updateTexture: function (src) {
     var data = this.data;
+    var self = this;
     var material = this.material;
     if (src) {
       if (src !== this.textureSrc) {
         // Texture added or changed.
         this.textureSrc = src;
         srcLoader.validateSrc(src, loadImage, loadVideo);
+        // Material is first to load this texture. Load and resolve texture.
       }
     } else {
       // Texture removed.
       material.map = null;
       material.needsUpdate = true;
     }
-    function loadImage (src) { loadImageTexture(material, src, data.repeat); }
-    function loadVideo (src) { loadVideoTexture(material, src, data.width, data.height); }
+
+    function loadImage (src) {
+      texturePromises[src] = loadImageTexture(material, src, data.repeat);
+      texturePromises[src].then(function (texture) {
+        self.el.emit('material-texture-loaded');
+      }); }
+    function loadVideo (src) {
+      texturePromises[src] = loadVideoTexture(material, src, data.width, data.height);
+      texturePromises[src].then(function (videoEl) {
+        videoEl.addEventListener('loadeddata', function (e) {
+          self.el.emit('material-texture-loaded', videoEl);
+        });
+        videoEl.addEventListener('ended', function (e) {
+          // works only for non-loop videos
+          self.el.emit('video-ended');
```

Comment:
A bit too magical. I'd be worried people making components will try to emit an event but wonder why their event is not firing because they didn't prepend the event name in the listener.